### ISSUE TITLE
Generate TUF files

### ISSUE THREAD

**Issue by zmanian** [heartsucker/rust-tuf#48]
Do you intended for this tool to be able to generate TUF files for a given binary in addition to verifying them?

Should we define an API for that?

**Comment by heartsucker** [comment]
Yes, I do intend on it eventually. However, you can do signing of metadata with like 6 lines of Python, and anything can generate JSON, so worst case scenario, this requires some outside tooling to be fully usable right now.

Signing and generating metadata depends on serialization #24 at least, and it wouldn't really make sense either until after we have delegations #16 too.

**Comment by heartsucker** [comment]
Work tracked in `rewrite` branch

**Comment by heartsucker** [comment]
Depends on #81 and possibly #82

**Comment by heartsucker** [comment]
Also, the groundwork for this is merged into `develop` already. Generation exists, but signing does not.

**Comment by heartsucker** [comment]
Signing tracked in (dirty) branch `sign`.
